hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/keygen/SimpleKeyGenerator.java

```diff
@@ -36,8 +37,9 @@
   private final SimpleAvroKeyGenerator simpleAvroKeyGenerator;
 
   public SimpleKeyGenerator(TypedProperties props) {
-    this(props, props.getString(KeyGeneratorOptions.RECORDKEY_FIELD_NAME.key()),
-        props.getString(KeyGeneratorOptions.PARTITIONPATH_FIELD_NAME.key()));
+    this(props,
```

Comment:
why return a default value. KeyGenerators are generally used in the write path where we take user records an generate record key and partition path and update the values to meta fields. This is counter-intuitive. can you help me understand this change.

Reply:
Right, changing the behavior of the default key gen is not desirable. Now that we are setting the metadata table properties like record key, we should always be reading the record with the right key for the metadata table. Also, introduced a custom key gen for the metadata table which doesn't dereference the partition filed from the record. Reverted all the changes pertaining to key gen.